processor/transformer/transformer.go

```diff
@@ -86,35 +94,48 @@ type HandleT struct {
 	receivedStat stats.Measurement
 	cpDownGauge  stats.Measurement
 
+	conf   *config.Config
 	logger logger.Logger
+	stat   stats.Stats
 
 	Client *http.Client
 
 	guardConcurrency chan struct{}
+
+	config struct {
+		maxConcurrency         int
+		maxHTTPConnections     int
+		maxHTTPIdleConnections int
+		disableKeepAlives      bool
+
+		timeoutDuration time.Duration
+
+		maxRetry int
+
+		failOnTimeout bool
+		failOnError   bool
+
+		destTransformationURL string
+		userTransformationURL string
+	}
+
+	warehouseDestinationMap map[string]struct{}
```

Comment:
Preparing `warehouseDestinationMap` from `warehouseutils.WarehouseDestinations` once instead of checking in slice every time for every transformations call.

Reply:
We might as well do this in an `init` function using a non-exported package-scoped variable